app/models/concerns/sortable_struct.rb

```diff
@@ -0,0 +1,36 @@
+module SortableStruct
+  extend ActiveSupport::Concern
+
+  included do
+    attribute :sort_direction, Types::SortDirection
+    attribute :sort_by, Types::String.default(default_sort_by).enum(*sortable_columns)
+  end
+
+  class_methods do
+    def sortable_columns
+      self::SORTABLE_COLUMNS
+    end
+
+    def default_sort_by
+      self::DEFAULT_SORT_BY
+    end
+
+    def default_sort_direction
+      self::DEFAULT_SORT_DIRECTION
+    end
+
+    def new_or_default(params = nil)
```

Comment:
Is this the best name?

Reply:
I think I'd like to refactor the controller where this is used so will take a look then.